test/MustSpendTotal.purs

```diff
@@ -0,0 +1,76 @@
+module Test.Ctl.MustSpendTotal
+  ( main
+  , suite
+  ) where
+
+import Prelude
+
+import Contract.TxConstraints
+  ( TxConstraints
+  , mustProduceAtLeast
+  , mustProduceAtLeastTotal
+  , mustSpendAtLeast
+  , mustSpendAtLeastTotal
+  )
+import Contract.Value as Value
+import Ctl.Internal.Test.TestPlanM (TestPlanM, interpret)
+import Data.Function (on)
+import Data.Newtype (class Newtype, unwrap)
+import Effect (Effect)
+import Effect.Aff (Aff, launchAff_)
+import Mote (group, test)
+import Test.QuickCheck (Result, (===))
+import Test.Spec.QuickCheck (quickCheck)
+
+-- Run with `spago test --main Test.Ctl.MustSpendTotal`
+main :: Effect Unit
+main = launchAff_ do
+  interpret suite
+
+suite :: TestPlanM (Aff Unit) Unit
+suite = do
```

Comment:
I think it would be better to have two tests for the `homomorphism`, one with a fixed name for the `currency symbol` and `token name` and the other with a mix of them. 

In fact I don't know if we need to test it with more than one `currency symbol` and `token name`.

Reply:
I do not understand what you mean by `one with a fixed name for the currency symbol and token name` and why it is better.

> In fact I don't know if we need to test it with more than one currency symbol and token name.

Why not? It does not produce complex or slow code, why creating special case instead of just using one Arbitrary instance?

Reply:
I thought it was slow, but otherwise is okay. 

The reason we would like to have two separate cases is to improve the possibility to have two different values with the same minting policy and token name in the test. I think this is the most interesting case and the one that needs more coverage from property testing.